src/intersections/triangles.jl

```diff
@@ -121,12 +121,73 @@ function intersecttype(s::Segment{3,T}, t::Triangle{3,T}) where {T}
       # rejection 1, coplanar segment
       return NoIntersection()
     end
-  end      
+  end
 
   λ = wᵥ / (wᵥ - sᵥ)
 
   # if λ is approximately 0 or 1, set as so to prevent any domain errors
   λ = isapprox(λ, zero(T), atol=atol(T)) ? zero(T) : (isapprox(λ, one(T), atol=atol(T)) ? one(T) : λ)
-  
+
   IntersectingSegmentTriangle(s(λ))
-end
\ No newline at end of file
+end
+
+
+"""
+    intersecttype(s, t)
+
+Calculate the intersection of a ray and triangle in 3D.
+
+## References
+
+* Möller, T. and Trumbore, B., 1997. [Fast, minimum storage ray-triangle
+  intersection. Journal of graphics tools]
+  (https://www.tandfonline.com/doi/abs/10.1080/10867651.1997.10487468)
+"""
+function intersecttype(r::Ray{3,T}, t::Triangle{3,T}) where {T}
```

Comment:
Can you please rebase the PR to the latest master? We have recently added a function as the first argument of `intersecttype` to escape type instability, see the latest comments here: https://github.com/JuliaGeometry/Meshes.jl/issues/54